raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -224,60 +226,70 @@ def get_pending_transfer_pairs(
     return pending_pairs
 
 
-def _fee_for_payer_channel(
-    channel: NettingChannelState, amount: PaymentWithFeeAmount
-) -> Optional[FeeAmount]:
-    """ Fee deducted by the mediator for an incoming channel.
+def find_intersection(fee_func: Interpolate, line: Callable[[int], float]) -> Optional[float]:
+    """ Returns the x value where both functions intersect
 
-    The `amount` is the total incoming amount without any fees deducted.
+    `fee_func` is a piecewise linear function while `line` is a straight line
+    and takes the one of fee_func's indexes as argument.
     """
-    balance = get_balance(channel.our_state, channel.partner_state)
-    try:
-        return channel.fee_schedule.fee_payer(amount, balance)
-    except UndefinedMediationFee:
-        return None
-
-
-def _fee_for_payee_channel(
-    channel: NettingChannelState, amount: PaymentWithFeeAmount
-) -> Optional[FeeAmount]:
-    """ Fee deducted by the mediator for an outgoing channel.
-
-    The `amount` is the incoming amount where the incoming fee is already
-    deducted, but the outgoing fee isn't (that's what this function does,
-    after all).
-    """
-    balance = get_balance(channel.our_state, channel.partner_state)
+    i = 0
+    y = fee_func.y_list[i]
+    comp = operator.lt if y < line(i) else operator.gt
```

Comment:
nitpick: Why not call `comp`, `compare`? Line is small anyway so no need to abbreviate.